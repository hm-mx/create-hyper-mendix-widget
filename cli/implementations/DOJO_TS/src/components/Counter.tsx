import React, { useState } from 'react';
import { ComponentProps } from 'typings/BaseProps';

const Counter = ({ style, dummyKey }: ComponentProps) => {
  const [count, setCount] = useState<number>(0);
  const down = () => setCount(count - 1);
  const up = () => setCount(count + 1);

  return (
    <div className="react-counter" style={style}>
      <div className="counter-header">
        <h1>{dummyKey}</h1>
      </div>
      <h1 className="counter-count">{count}</h1>
      <div className="controls-wrapper">
        <button type="button" className="counter-btn" onClick={down}>
          -
        </button>
        <button type="button" className="counter-btn" onClick={up}>
          +
        </button>
      </div>
    </div>
  );
};

export default Counter;
