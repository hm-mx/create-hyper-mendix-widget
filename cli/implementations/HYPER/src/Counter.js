import { h, app } from 'hyperapp';

export default function(widgetProps,mountingDomNode) {
	const state = {
		count: 0,
		...widgetProps
	};
	
	const actions = {
		down: (value) => (state) => ({ count: state.count - value }),
		up: (value) => (state) => ({ count: state.count + value })
	};
	
	const Counter = (state, actions) => (
		<div className="hyperapp-counter">
			<div className="counter-header">
				<span>•••</span>
				<h1>{state.dummyKey}</h1>
				<span>•••</span>
			</div>
			<h1 className="counter-count">{state.count}</h1>
			<div className="controls-wrapper">
				<button className="counter-btn" onclick={() => actions.down(1)}>
					-
				</button>
				<button className="counter-btn" onclick={() => actions.up(1)}>
					+
				</button>
			</div>
		</div>
	);
	app(state, actions, Counter, mountingDomNode);
}
