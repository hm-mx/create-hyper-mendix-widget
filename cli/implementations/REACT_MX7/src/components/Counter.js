import React from "react";

class Counter extends React.Component {
  constructor(props) {
    super(props);
    this.state = {count: 0};
    this.up = this.up.bind(this);
    this.down = this.down.bind(this);
  }
  down() {
    this.setState(state => ({count: state.count - 1}));
  }
  up() {
    this.setState(state => ({count: state.count + 1}));
  }
  render() {
    return (
      <div className={`react-counter ${this.props.class}`} style={this.props.style}>
        <div className="counter-header">
          <span>•••</span>
          <h1>
            {this.props.dummyKey}
          </h1>
          <span>•••</span>
        </div>
        <h1 className="counter-count">
          {this.state.count}
        </h1>
        <div className="controls-wrapper">
          <button className="counter-btn" onClick={this.down}>
            -
          </button>
          <button className="counter-btn" onClick={this.up}>
            +
          </button>
        </div>
      </div>
    );
  }
}

export default Counter;
