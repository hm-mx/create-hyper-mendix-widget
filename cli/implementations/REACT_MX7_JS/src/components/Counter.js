import React from 'react';

class Counter extends React.Component {
  state = { count: 0, isReady: false };

  /**
   * in case your widget requires context
   * i.e. needsEntityContext="true" in `widget.config.ejs`
   */
  componentDidUpdate() {
    const { mxObject } = this.props;
    if (mxObject) {
      this.setState({ isReady: true });
    }
  }

  down = () => {
    this.setState(state => ({ count: state.count - 1 }));
  };

  up = () => {
    this.setState(state => ({ count: state.count + 1 }));
  };

  render() {
    const { count } = this.state;
    const { style, dummyKey } = this.props;

    return (
      <div className="react-counter" style={style}>
        <div className="counter-header">
          <h1>{dummyKey}</h1>
        </div>
        <h1 className="counter-count">{count}</h1>
        <div className="controls-wrapper">
          <button type="button" className="counter-btn" onClick={this.down}>
            -
          </button>
          <button type="button" className="counter-btn" onClick={this.up}>
            +
          </button>
        </div>
      </div>
    );
  }
}

export default Counter;
