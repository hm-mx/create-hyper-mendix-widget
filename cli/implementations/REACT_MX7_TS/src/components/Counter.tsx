import React from 'react';

interface CounterProps {
  style?: { [key: string]: string };
  dummyKey: string;
  mxObject?: {
    get: Function;
    set: Function;
    getGuid: Function;
  };
}

interface CounterState {
  count: number;
}

class Counter extends React.Component<CounterProps, CounterState> {
  state = { count: 0 };

  /**
   * in case your widget requires context
   * i.e. needsEntityContext="true" in `widget.config.ejs`
   */
  componentDidMount() {
    const { mxObject } = this.props;
    if (mxObject) {
      // Do something one time when component is rendered
    }
  }

  componentDidUpdate() {
    const { mxObject } = this.props;
    if (mxObject) {
      // Do something that will fire every time when context is changed
    }
  }

  down = () => {
    this.setState(({ count }) => ({ count: count - 1 }));
  };

  up = () => {
    this.setState(({ count }) => ({ count: count + 1 }));
  };

  render() {
    const { count } = this.state;
    const { style, dummyKey } = this.props;

    return (
      <div className="react-counter" style={style}>
        <div className="counter-header">
          <h1>{dummyKey}</h1>
        </div>
        <h1 className="counter-count">{count}</h1>
        <div className="controls-wrapper">
          <button type="button" className="counter-btn" onClick={this.down}>
            -
          </button>
          <button type="button" className="counter-btn" onClick={this.up}>
            +
          </button>
        </div>
      </div>
    );
  }
}

export default Counter;
