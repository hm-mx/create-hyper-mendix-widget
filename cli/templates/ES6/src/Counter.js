export default function (widgetProps, mountingNode) {

    //  Counter Component wrapper
    const counterWrapper = document.createElement("div");
    counterWrapper.className = "es6-counter";

    // Header
    const counterHeader = document.createElement("div");
    counterHeader.className = "counter-header";

    const spanLeft = document.createElement("span");
    spanLeft.innerText = "•••";

    const spanRight = document.createElement("span");
    spanRight.innerText = "•••";

    const headerText = document.createElement("h1");
    headerText.innerText = widgetProps.dummyKey;

    counterHeader.appendChild(spanLeft);
    counterHeader.appendChild(headerText);
    counterHeader.appendChild(spanRight);


    // Counter Count
    const counterCount = document.createElement("h1");
    counterCount.className = "counter-count";
    counterCount.innerText = '0';


    // Controls wrapper
    const controlsWrapper = document.createElement("div");
    controlsWrapper.className = "controls-wrapper";

    const btnDown = document.createElement("button");
    btnDown.className = "counter-btn";
    btnDown.innerText = "-";


    const btnUp = document.createElement("button");
    btnUp.className = "counter-btn";
    btnUp.innerText = "+";


    controlsWrapper.appendChild(btnDown);
    controlsWrapper.appendChild(btnUp);


    counterWrapper.appendChild(counterHeader);
    counterWrapper.appendChild(counterCount);
    counterWrapper.appendChild(controlsWrapper);


    // mount counter to the dom
    mountingNode.appendChild(counterWrapper);


    // set actions
    btnUp.onclick = () => {
        const nextCount = parseInt(counterCount.innerText, 10) + 1;
        counterCount.innerText = nextCount;
    };
    btnDown.onclick = () => {
        const nextCount = parseInt(counterCount.innerText, 10) - 1;
        counterCount.innerText = nextCount;
    };


};