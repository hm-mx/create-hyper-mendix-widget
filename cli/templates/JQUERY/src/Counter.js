import $ from "jquery";

export default function (widgetProps, mountingNode) {


    //  Counter Component wrapper
    const counterWrapper = $("<div></div>").addClass("jquery-counter");


    // Header
    const counterHeader = $("<div></div>").addClass("counter-header")
        .append($("<span></span>").text("•••"))
        .append($("<h1></h1>").text(widgetProps.dummyKey))
        .append($("<span></span>").text("•••"));


    // Counter Count
    const counterCount = $("<h1></h1>").addClass("counter-count").text("0");

    // Controls wrapper
    const controlsWrapper = $("<div></div>").addClass("controls-wrapper")
        .append($("<button></button>").addClass("counter-btn").text("-").click(e => {
            const counter = $(counterWrapper).find("h1.counter-count");
            counter.text(parseInt(counter.text(), 10) - 1);
        }))
        .append($("<button></button>").addClass("counter-btn").text("+").click(e => {
            const counter = $(counterWrapper).find("h1.counter-count");
            counter.text(parseInt(counter.text(), 10) + 1);
        }));


    $(counterWrapper)
        .append(counterHeader)
        .append(counterCount)
        .append(controlsWrapper);



    // mount counter to the dom
    $(mountingNode).append(counterWrapper);



};