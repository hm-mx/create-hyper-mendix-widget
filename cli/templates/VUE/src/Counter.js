import Vue from "vue";

const Counter = ({
    dummyKey
}, mountingNode) => (new Vue({
    el: mountingNode,
    template: `
            <div class="vue-counter">
                <div class="counter-header">
                    <span>•••</span>
                    <h1>{{dummyKey}}</h1>
                    <span>•••</span>
                </div>
                <h1 class="counter-count">{{count}}</h1>
                <div class="controls-wrapper">
                    <button class="counter-btn" @click="down">
                        -
                    </button>
                    <button class="counter-btn" @click="up">
                        +
                    </button>
                </div>
            </div>
    `,
    data: {
        count: 0,
        dummyKey
    },
    methods: {
        up() {
            this.count++;
        },
        down() {
            this.count--;
        }
    }
}));
export default Counter;