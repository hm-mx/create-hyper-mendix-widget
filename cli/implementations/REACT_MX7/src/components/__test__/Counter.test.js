import React from 'react';
import renderer from 'react-test-renderer';

import Counter from '../Counter';

describe('Counter Component Test Unit', () => {
  it('should render correctly', () => {
    const props = { dummyKey: 'SOME_DUMMY_KEY' };
    const CounterSnapshot = renderer.create(<Counter {...props} />).toJSON();
    expect(CounterSnapshot).toMatchSnapshot();
  });
});
