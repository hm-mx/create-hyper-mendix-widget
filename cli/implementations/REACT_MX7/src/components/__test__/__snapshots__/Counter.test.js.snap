// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component Test Unit should render correctly 1`] = `
<div
  className="react-counter undefined"
>
  <div
    className="counter-header"
  >
    <span>
      •••
    </span>
    <h1>
      SOME_DUMMY_KEY
    </h1>
    <span>
      •••
    </span>
  </div>
  <h1
    className="counter-count"
  >
    0
  </h1>
  <div
    className="controls-wrapper"
  >
    <button
      className="counter-btn"
      onClick={[Function]}
    >
      -
    </button>
    <button
      className="counter-btn"
      onClick={[Function]}
    >
      +
    </button>
  </div>
</div>
`;
